WooCommerce/Classes/Tools/Currency/CurrencyFormatter.swift

```diff
@@ -110,7 +110,13 @@ public class CurrencyFormatter {
     ///     - currency: a 3-letter country code for currencies that are supported in the API. e.g. "USD"
     ///
     func formatAmount(_ stringAmount: String, with currency: String = CurrencySettings.shared.currencyCode.rawValue) -> String? {
-        guard let decimalAmount = convertToDecimal(from: stringAmount) else {
+
+        // NSDecimalNumber use by default the local decimal separator to evaluate a decimal amount.
+        // We substitute the current decimal separator with the locale decimal separator.
+        let localeDecimalSeparator = Locale.current.decimalSeparator ?? CurrencySettings.shared.decimalSeparator
```

Comment:
This should be reversed. The phone's locale should never be used as the default, only as a last option.

Reply:
You are right, but in this case I explicit want to use the locale as default, just to make sure that NSDecimalNumber is initialized correctly since it uses the locale decimal separator. 
For example, if we call formatAmount() on a website that is using the comma as decimal separator, and in our locale we use the point `.` we have this case:
Input value of **NSDecimalNumber**: `10,99`
Output value of **NSDecimalNumber**: `10.00`
Expected value: `10.99`

Reply:
Oh UGH. I just re-read the documentation. I see what you mean. NSDecimalNumber assumes the current locale when converting a string value. And because the locale isn't known from our website Settings API, we can't init an NSDecimalNumber with the correct locale (which is how it would choose the correct decimal separator).